MiniAn/minian_main.py

```diff
@@ -431,7 +432,112 @@ def load_doric_to_xarray(
 
     return varr, file_
 
-
+            
+def cross_register(minian_parameters, currentFile_AC, currentFile_A):
+    performCrossReg = minian_parameters.params_crossRegister["crossReg"]
+    if performCrossReg:
+        param_dist = 5 # Defines the max dist between centroids on different sessions to consider them as same cell
+        refFileName = minian_parameters.params_crossRegister["fname_crossReg"]
+        refImages   = minian_parameters.params_crossRegister["h5path_images"]
+        refRois     = minian_parameters.params_crossRegister["h5path_roi"]
+
+        # Load base/reference file to xarray
+        refCR, fileCR_ =  mn_utils.load_doric_to_xarray(refFileName, refImages, np.float64, None, "subset", None)
+
+        # max project of the miniAn images for base file and current file
+        ref_max     = refCR.max("frame")
+        current_max = currentFile_AC.max("frame")
+
+        result = xr.concat([ref_max, current_max], pd.Index(['session1', 'session2'], name = "session"))
+        temps = result.rename('temps')
+        # chk, _ = get_optimal_chk(temps, **minian_parameters.params_get_optimal_chk)
+        # temps = temps.chunk({"frame": 1, "height": -1, "width": -1}).rename("temps")
+
+        # estimate shift 
+        shifts = estimate_motion(temps, dim = 'session').compute().rename('shifts')
+
+        # Apply Shifts
+        temps_sh = apply_transform(temps, shifts).compute().rename('temps_shifted')
+        shiftds = xr.merge([temps, shifts, temps_sh])
+
+        window    = shiftds['temps_shifted'].isnull().sum('session')
+        window, _ = xr.broadcast(window, shiftds['temps_shifted'])
+
+        # function to get ROI footprints ('A') for the base (reference) file
+        refFootprints = getRefFileFootprints(refFileName, refRois, refCR.coords)
+
+        # change the unitIds of the current file footprints to number starting from max UnitId in ref file
+        updatedUnitIds = []
+        refUnitIDMax = (refFootprints.coords["unit_id"].values).max() + 1
```

Comment:
For variable and function in python can you name them using Snake Case convention (using _ to separet words) for example `ref_unit_id_max` .  And for variable definition do like in qt using all upper case.

In Qt we use the Camel case for variable and function name and UpperCamelCase for class

For information some [naming conventions](https://www.pluralsight.com/blog/software-development/programming-naming-conventions-explained#:~:text=snake_case%20is%20a%20variable%20naming,database%20table%20and%20column%20names. )